### PR TITLE
Create issue templates

### PR DESCRIPTION
**Pull request by luanpotter**
In order to keep things better organize and optimize the help the community can provide, I am proposing to create some issue templates with strict rules. These rules might be enforced if necessary by closing new non-compliant issues. For existing issues, we will try to keep or port them but we might if necessary, give a 2-week notice that they will be closing for not following the rules. If after 2 weeks they have not been updated, they can be closed as well.

I also changed the README a little bit to reflect these rules and our expectations around: questions, bugs, feature requests and PRs, pointing to a new contributing doc.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+name: Bug Report
+about: If you found a bug, please use this template to report it.
+title: ''
+labels: bug
+assignees: ''
+
+---
+
+["one liner" - quickly and concisely describe your bug with one very brief paragraph, preferably between 1-3 lines, in clear, correct and acceptable English]
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:10]: Remove "quickly and" imho

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+**Full Description**
+Describe in full what the bug is, exactly how, what and when it happens. In particular specifying what you thought should happen and what did happen, and how they differ, in detail. Unless the bug is abundantly clear by the "one liner" (rarely is), this is mandatory. Try to be concise but err on the side of more, not less.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:13]: In full -> thoroughly
Err on the side of more, not less -> rather write too much than too little

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+**Code to Reproduce**
+A [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), i.e., code that allows us to replicate the bug. This is **mandatory**. It can be one line, if it already is enough to reproduce (rarely is), or a bigger example with all your relevant code. You **must** use code blocks to put your code in. Don't send your whole project repo, please read the article linked and extract a **minimum** example, but feel free to link a newly created sample repo reproducing the issue instead of pasting code if necessary (rarely is). If you do link a repo, you must put in at least a little bit of the relevant code and where to find it on the repo.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:16]: I would remove both of the "(rarely is)"

Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:16]: Minimum -> minimal

Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:16]: On -> In the repo

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+**Log Errors**
+If your problem involves log errors or messages, please put them here, in full, but feel free to highlight the parts that are relevant. You **must** use code blocks (or gists) to paste in log lines. You can use gists to keep things organized. An absolutely outstanding example would be to add a code block with 2-3 relevant log lines, followed by a link to a gist with the whole log from witch the relevant lines were extracted.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:19]: in full -> in full. Feel free...
I think the sentence "You can use gists to keep things organized." can be removed.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+**Files/URLs/Sources**
+If your issue or bug involves specific files, you must provide them. If they are URLs they should already be somewhere in the code provided but please replicate those here. If they are private URLs or actual files, please upload them somewhere accessible and add them here. If your issue is with a stream, for example, provide the stream link or if the stream is private, create a similar stream that also causes the problem and is public, and add that. If you don't know if your issue involves specific resources, please download a few of the [mp3 sample files](https://github.com/luanpotter/audioplayers/tree/master/example/assets) present in the `example` app and test with those instead of whatever you are doing currently. If the issue persists, it does not involve specific files. Otherwise, this section is **mandatory**.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:22]: Remove ", for example".

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+**Screenshots**
+If applicable, add screenshots or video recordings to help explain your problem. This is totally optional.
+
+**Platforms**
+List in detail here **all** the platforms that you have tested, indicating if the error occurs or not in witch ones. Please try to test in as many different platforms as possible to make our lives easier (this is optional). You must add at least one platform in which the problem occurs (evidently). For each platform listed, you must specify:
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:28]: Witch -> which

Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:28]: "Indicating which platforms the error occurs on"

In as many -> on as many

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+* OS: web, android, ios, macos, etc. if web, which browser was used.
+* OS version: the android version, ios/macos versions or browser version used.
+* Device: whether it was a physical device or emulator/simulator. if running on physical device, what brand/model/kind of device was used. If running on emulator, what host desktop was used. for web/desktop, what desktop OS and version was used.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:32]: if -> If
On -> on a/an (two places)
for -> For

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+* flutter version: what exact flutter version you used. if you think it might be a configuration/setup issue, feel free to add a link to a gist with the output of `flutter doctor -v` (this is optional).
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:33]: if -> If

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+* audioplayers version: what version on pub you used, or if you built it yourself what branch and was the latest commit hash
+* release or not release: specify if you ran the app in release, profile or debug modes. also specify here any additional flags or conditions that might be relevant (for example, if you ran flutter web with the skia use canvas flag set to true, please mention that here).
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/bug-report.md:35]: also -> Also

**Review by spydon — COMMENTED** [review]
Just some grammar, but it looks good overall!

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+name: Feature Request
+about: If you have a suggestion for some change or new feature, please use this template
+  to report it.
+title: ''
+labels: feature-request
+assignees: ''
+
+---
+
+*First of all, make sure if the feature you are proposing doesn't already exist. Delete this line only once you did that.*
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:11]: if -> that

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+["one liner" - quickly and concisely describe your feature request with one very brief paragraph, preferably between 1-3 lines, in clear, correct and acceptable English]
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:13]: Remove "quickly and"

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+**Full Description**
+Describe in full what the feature you desire is, why it's useful and why it fits in this package (and not a standalone package for example). Mention what is your use case (what you want to use it for) and how it should be included (what API changes are necessary, if they are breaking changes, etc). This is mandatory. Please include how it should be implemented as well (is it a new flag to pass through to android/ios, is it just a new method using some build in functions, etc). This last part is optional but highly encouraged. It's super valuable to do some research to help us make sure your idea is doable. This section is mandatory.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:16]: "what is your use case" -> "what your use case is"
"what API changes are necessary" -> "what API changes that are necessary"

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+**Example Code**
+Unless your feature requires no API changes whatsoever, provide a very concise code example of how it would be used in a Flutter app. This is **mandatory** if your feature requires any API changes. You **must** use code blocks to put your code in. You can link an example repo (unnecessary) but you must add a simple version on the issue.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:19]: Remove "(unnecessary)", already implied as optional by "can"

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+**Platforms**
+Make sure to identify in which platforms this must be implemented. By platforms here I mean: android, ios, web, macos, etc. This section is mandatory.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:22]: in -> for

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+
+First, assess if this already exists or not on some platforms but not the others. Some stuff we currently only support on Android or only on iOS because Open Source projects are made with the help of the community and we not always have time to do everything. If, for example, you want an iOS feature to be added to android, that is totally fine but specify that here. In this case, provide in the sections above clearly explanations of how it already works on platform all platforms its present (eg the Code section must include code for how to use it on platform X and what changes (if any, probably none) are necessary on the api level to add it to platform Y).
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:24]: "not the other" -> "not on others"

Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:24]: "In this case, provide in the sections above clearly explanations" -> "In this case provide a clear explanation in the section above"

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+Secondly, asses in which platforms this make sense to be added. Specially web/macos. There are some features that just don't make sense outside of the mobile world, for example. Not everything needs to be present on all platforms. Please think about which platforms this should be added for, ideally. Doesn't mean we will end up doing all of them to start, but just as the goal.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:26]: Specially -> Especially for web/macos

Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:26]: Doesn't -> That doesn't

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+
+This is also a perfect section to include, if your prior research already reveled such a thing, if there are minimum OS versions that are absolutely required to add this feature. For example, if it's only possible to do it after Android 23, add that detail here.
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/feature-request.md:28]: reveled -> revealed
remove "such a thing, "

File: contributing.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+
+Thanks for using audioplayers and specially for your interest in contributing to the community!
+
+Please read this document carefully before doing anything else!
+
+## Old Issues/PRs
+
+We have many existing open issues and a few open PRs that were created before this doc was created. We will try to respect their innocence of this file existence by doing our best effort to answer/address/fix/merge them as we normally would up to this point (i.e. as time permits). However, if an existing issue or PR is too blatant of an outlier from these rules, we reserve the right of asking, in the issue/PR for the author (or someone) to fix it so that it falls under the new rules (i.e. apply the templates, etc). If we need to do that, we will give two weeks for the issue/PR to be updated to follow the rules, otherwise it will be closed.
+
+Of course anyone is free to open a similar followup as long as the new one follows the rules at any time.
```
Comment (spydon) [inline contributing.md:11]: Remove "at any time"

File: contributing.md [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+
+## Contribution Types
+
+After you read and accepted the rules above, you need to decide what kind of inquiry do you have. Choose the most appropriate of the sections below.
+
+### Questions
+
+Questions are not bugs! Do not open issues for questions. Here are the channels to ask for help.
+
+First of all, make sure you read *at least* the [Readme Document](README.md) in full. That is the basis of how this library work and its very well written with care an love by us. If you haven't read even the readme, don't expect us to answer a question that is already solved there.
```
Comment (spydon) [inline contributing.md:34]: an -> and

File: contributing.md [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+Second, make sure you went through our FAQ, [Troubleshooting](troubleshooting.md). There are many questions we get asked all the time that we have put the time and effort to answer on that doc. So make sure your question is not already there.
+
+Third, if your question is not there, try to searching for old issues. Even though we no longer user issues to track questions, we used to have A LOT of questions in the closed issues that are still there for documentation. Also maybe an old issue might give you context on how something works. In fact, if that doesn't work, try just searching google, stack overflow, discord logs. Maybe your question was already answered!
```
Comment (spydon) [inline contributing.md:38]: user -> use

File: contributing.md [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
+Lastly, try finding the answer in the source code. We try to keep our code clean and easy to understand, including docs explaining how things work. If your code/docs are not so clear, this is a great opportunity to help with a PR (see feature requests below).
```
Comment (spydon) [inline contributing.md:40]: your -> our ?
"are not so clear" -> "are clear on something"

File: contributing.md [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+
+ * The `flame` tag on [Stack Overflow](https://stackoverflow.com/questions/tagged/flame): Since audioplayers is part of the flame project, feel free to use the [flame] tag on Stack Overflow to get people from the community to help. This might be a bit more involved than discord but if you make a properly acceptable Stack Overflow question, people will be much more willing to help you with hard problems. Also, you are leaving some documentation for future generations!
+
+### Bugs / Issue Reports
+
+If you found a bug or issue, please report it to us! If you are unsure if it's a bug or a question, feel free to ask on the discord channel first, or if you truly believe in good faith it's a bug, you can open an issue on GitHub.
+
+But the first step is, again, search for an existing issue. Maybe your issue was already reported, and we don't want duplicates. In fact, if it was already reported, the existing issues might have tips and tricks to circumvent the issue until we fix it.
```
Comment (spydon) [inline contributing.md:52]: to search for an existing issue

File: contributing.md [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+
+Once you are secure your bug is brand new, you can create an issue and select the `Bug Report` type. You **must** follow the template provided, read it carefully.
```
Comment (spydon) [inline contributing.md:54]: secure -> certain

File: contributing.md [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+
+### Feature Requests / PRs
+
+Unless your PR is super simple (i.e. typo fixes, documentation improvements, etc), please open a Feature Request issue before opening a PR. You can make it clear on your feature request that you are willing to contribute with a PR, but it's important to have some discussion before starting anything more complicated; we might have better suggestions of how to do things.
```
Comment (spydon) [inline contributing.md:58]: "on your feature request" -> "in your feature request"

File: contributing.md [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+
+In order to open a Feature Request issue, just select the correct template under issue creation. You **must** follow the guidelines in the template.
+
+Once your feature got approved to start developing, feel free to send your PRs! However, we have a few important PR rules:
+
+ * Your build must pass. Please make sure everything is green!
+ * Follow guidelines. We don't have a code analyzer for the native side (yet!), but please follow the code around you to make it properly formatted and linted. For Java, please follow an acceptable standard [like this one](https://google.github.io/styleguide/javaguide.html). There is nothing worse than badly formatted code!
+ * Write clean, beautiful and easy to understand code, with comments if necessary and docs if possible.
+ * Update our README/docs accordingly to your change, making it clear witch platforms are supported.
```
Comment (spydon) [inline contributing.md:67]: witch -> which

File: contributing.md [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+ * Try to support all platforms that it makes sense. This is a hard ask, and we understand and we will merge PRs that only work on one platform. But if you have the time, please help us with feature parity.
```
Comment (spydon) [inline contributing.md:68]: that it makes sense -> where it makes sense
ask -> thing to do
on one platform -> on only one platform as well

**Review by spydon — COMMENTED** [review]
A few more suggested grammar changes

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 13 above)
Comment (luanpotter) [reply .github/ISSUE_TEMPLATE/feature-request.md:19]: I wanted to emphasize it's not necessary at all.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (luanpotter) [reply .github/ISSUE_TEMPLATE/bug-report.md:16]: I wanted to emphasize that it's most often not the case to force people to think twice on those options.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (luanpotter) [reply .github/ISSUE_TEMPLATE/bug-report.md:10]: I meant shortly, not quickly, but that is redundant with concise

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 5 above)
Comment (luanpotter) [reply .github/ISSUE_TEMPLATE/bug-report.md:22]: but it is an example :p

**Review by spydon — APPROVED** [review]
Much better! :)

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**